### ISSUE TITLE
[CI/CD] Move Prettier formatting checks to a GitHub action

### ISSUE THREAD

**Issue by Atmosfearful** [KlimaDAO/klimadao#1438]
In the klimadao monorepo I would like to set up a GHA that runs `prettier --check` on every commit and throws a red status if an issue is surfaced.

# background
To reduce build times and simplify our dependency tree I removed prettier checks from all packages and added a `prettier-check-all` command to the top level of npm workspaces`. I'd like to bring it back in the form of a parallel GHA instead of running it before each build.

# Acceptance criteria:
- [ ] Every commit to every branch in the `klimadao` monorepo should trigger a `check formatting` status check
- [ ] It should run in parallel with the other actions & bots (vercel)
- [ ] the status check should throw a red X when `prettier check` catches something

## Stretch goals
Bonus points if you can only check directories with changes i.e. with `git diff`
  1. get list of top level dirs containing changes with `git diff --name-only [TARGET_BRANCH] | awk -F'/' '{print $1}' | sort -u`
    - for `staging`, check against `main`
    - for branches that are not `staging` or `main`, i.e. any working branch, check against `staging`
  2. pass this list of dirs as an argument to `prettier --check` so that only changed files are checked.

**Comment by 0xAurelius** [comment]
Got it, can do
